From 09c1e33f841be997a582cd20b016a63c72d6c231 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 1 Feb 2024 12:22:11 +0100
Subject: [PATCH 1/2] widget: Make the icon spacing configurable

Having value 4 is too large for the default panel with size 24px, but value 0 would be too small for a larger panel. Add a new "panel-icon-spacing" setting to allow users to configure for their needs.

https://gitlab.gnome.org/GNOME/gnome-panel/-/issues/20
---
 data/org.gnome.gnome-panel.gschema.xml |  7 +++++++
 gnome-panel/gp-application.c           | 25 ++++++++++++++++++++++---
 gnome-panel/panel-widget.c             | 15 +++++++--------
 3 files changed, 36 insertions(+), 11 deletions(-)

diff --git a/data/org.gnome.gnome-panel.gschema.xml b/data/org.gnome.gnome-panel.gschema.xml
index 08c4bbfa3..10a322282 100644
--- a/data/org.gnome.gnome-panel.gschema.xml
+++ b/data/org.gnome.gnome-panel.gschema.xml
@@ -58,6 +58,13 @@
       <description>The maximum size of icons in panels.</description>
     </key>
 
+    <key name="panel-icon-spacing" type="u">
+      <range min='0' max='32'/>
+      <default>2</default>
+      <summary>Icon spacing in panels</summary>
+      <description>On horizontal panel minimum space between top and bottom border and the icon. For vertical panel minimum space from left and right borders.</description>
+    </key>
+
     <key name="theme-variant" enum="org.gnome.gnome-panel.PanelThemeVariant">
       <default>'system'</default>
       <summary>Which theme variant to use</summary>
diff --git a/gnome-panel/gp-application.c b/gnome-panel/gp-application.c
index 4ff5fc598..63850b5a1 100644
--- a/gnome-panel/gp-application.c
+++ b/gnome-panel/gp-application.c
@@ -297,9 +297,7 @@ set_panel_icon_size (AppletInfo *applet,
 }
 
 static void
-panel_max_icon_size_cb (GSettings     *settings,
-                        const char    *key,
-                        GpApplication *self)
+update_icon_size (GpApplication *self)
 {
   GHashTableIter iter;
   gpointer value;
@@ -317,6 +315,22 @@ panel_max_icon_size_cb (GSettings     *settings,
     }
 }
 
+static void
+panel_max_icon_size_cb (GSettings     *settings,
+                        const char    *key,
+                        GpApplication *self)
+{
+  update_icon_size (self);
+}
+
+static void
+panel_icon_spacing_cb (GSettings     *settings,
+                       const char    *key,
+                       GpApplication *self)
+{
+  update_icon_size (self);
+}
+
 static gboolean
 initable_init (GInitable     *initable,
                GCancellable  *cancellable,
@@ -413,6 +427,11 @@ gp_application_init (GpApplication *self)
                     G_CALLBACK (panel_max_icon_size_cb),
                     self);
 
+  g_signal_connect (self->general_settings,
+                    "changed::panel-icon-spacing",
+                    G_CALLBACK (panel_icon_spacing_cb),
+                    self);
+
   g_signal_connect (self->interface_settings,
                     "changed::color-scheme",
                     G_CALLBACK (theme_variant_changed_cb),
diff --git a/gnome-panel/panel-widget.c b/gnome-panel/panel-widget.c
index 42121fb1a..23c2a2c18 100644
--- a/gnome-panel/panel-widget.c
+++ b/gnome-panel/panel-widget.c
@@ -2713,21 +2713,20 @@ panel_widget_get_icon_size (PanelWidget *self)
   general_settings = gp_application_get_general_settings (application);
   panel_max_icon_size = g_settings_get_enum (general_settings,
                                              "panel-max-icon-size");
-
-  spacing = 4;
+  spacing = g_settings_get_uint (general_settings, "panel-icon-spacing");
   panel_size = self->sz;
 
-  if (panel_size <= panel_max_icon_size + spacing)
+  if (panel_size <= panel_max_icon_size + spacing * 2)
     {
-      if (panel_size < 22 + spacing)
+      if (panel_size < 22 + spacing * 2)
         panel_icon_size = 16;
-      else if (panel_size < 24 + spacing)
+      else if (panel_size < 24 + spacing * 2)
         panel_icon_size = 22;
-      else if (panel_size < 32 + spacing)
+      else if (panel_size < 32 + spacing * 2)
         panel_icon_size = 24;
-      else if (panel_size < 48 + spacing)
+      else if (panel_size < 48 + spacing * 2)
         panel_icon_size = 32;
-      else if (panel_size < 64 + spacing)
+      else if (panel_size < 64 + spacing * 2)
         panel_icon_size = 48;
       else
         panel_icon_size = 64;
-- 
GitLab


From 1fd50b3f4a656681529e714e55c8fab6bb689516 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Tue, 27 Feb 2024 00:46:20 +0100
Subject: [PATCH 2/2] widget: Reduce icon spacing to zero by default

The height of the default panel is only 24px, so no much space left for spacing.

https://gitlab.gnome.org/GNOME/gnome-panel/-/issues/20
---
 data/org.gnome.gnome-panel.gschema.xml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/org.gnome.gnome-panel.gschema.xml b/data/org.gnome.gnome-panel.gschema.xml
index 10a322282..26d2360e6 100644
--- a/data/org.gnome.gnome-panel.gschema.xml
+++ b/data/org.gnome.gnome-panel.gschema.xml
@@ -60,7 +60,7 @@
 
     <key name="panel-icon-spacing" type="u">
       <range min='0' max='32'/>
-      <default>2</default>
+      <default>0</default>
       <summary>Icon spacing in panels</summary>
       <description>On horizontal panel minimum space between top and bottom border and the icon. For vertical panel minimum space from left and right borders.</description>
     </key>
-- 
GitLab

